File: 169. Majority Element/169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        d = {}
        for i in nums:
            if (i in d):
                d[i]+=1
            else:
                d[i] = 1
        maxnum = 0
        
        for k, v in d.items():
            if (v > maxnum):
                maxnum = v
                re = k
        return re


solution 2

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        re = len(nums)/2
        return nums[int(re)]

File: 169. Majority Element/169.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        var d = new Dictionary<int, int>();
        
        foreach (var i in nums){
            if (d.ContainsKey(i))
                d[i]++;
            else{
                d.Add(i, 1);
            }
        }
        
        foreach(KeyValuePair<int,int> kvp in d){
            if (kvp.Value > nums.Length/2)
                return kvp.Key;
        }
        
        return -1;
    }
}